core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -2041,6 +2044,20 @@ private void parseFromClause(GenericLexer lexer, QueryModel model, QueryModel ma
             }
         }
 
+        if (tok != null && isPivotKeyword(tok)) {
+            lexer.unparseLast();
```

Comment:
this is chaotic parsing, we already know this is "PIVOT" why does `parsePivot` method have to check this again?